src/python/tests/core/compare_core_model.py

```diff
@@ -0,0 +1,75 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Check the current ensembl-py core model against a core created from Ensembl SQL.
+This script gets one row for each table in the ORM to check that SQLalchemy can correctly query the table.
+If not, it will show the OperationalError exception to explain what is wrong in the ORM.
+
+Use this script this check the ORM and fix it.
+"""
+
+import logging
+
+from ensembl.utils.database import DBConnection
+from ensembl.utils.argparse import ArgumentParser
+from ensembl.utils.logging import init_logging_with_args
+from sqlalchemy import select
+from sqlalchemy.exc import NoResultFound, MultipleResultsFound, OperationalError, ProgrammingError
+from sqlalchemy.orm import Session
+
+from ensembl.core.models import Base
+
+def check_tables(session: Session, only_table: str = "") -> None:
+    success = []
+    errors = []
+    for table_name, table in Base.metadata.tables.items():
+        if isinstance(table_name, tuple):
+            table_name = table_name[0]
+        if only_table and table_name != only_table:
+            continue
+        logging.debug(f"Check table {table_name}")
+        stmt = select(table)
+        try:
+            session.execute(stmt).one()
+            success.append(table_name)
+        except (NoResultFound, MultipleResultsFound):
+            success.append(table_name)
+            pass
+        except (OperationalError, ProgrammingError) as err:
+            # Show the problematic query and continue
+            logging.warning(f"{table_name}: {err}")
+            errors.append(table_name)
+    
+    logging.info(f"{len(success)} tables successfully queried with the ORM")
+    if errors:
+        logging.warning(f"{len(errors)} tables failed to be queried with the ORM: {", ".join(errors)}")
+    else:
+        logging.info("No errors found")
+
+def main() -> None:
+    """Main script entry-point."""
+    parser = ArgumentParser(
+        description=__doc__
+    )
+    parser.add_argument("--url", required=True, type=str, help="MySQL URL to a core database to get tables data from")
```

Comment:
I would suggest to add the server arguments instead, as it allows better flexibility:
```suggestion
    parser.add_server_arguments(database=True, help="Ensembl MySQL core database")
```

Reply:
I completely forgot: there is `add_argument_url()` that will generate an SQLAlchemy URL object directly.

Trying to have both options available (url + server args) at the same time is possible, but may have a significant constrain (because mutually exclusive groups can only be defined at the main parser level).

Reply:
NB: I have switched to using the add_server_arguments. If we add --url it would be propagated here anyway